java/src/jmri/jmrit/beantable/BlockTableAction.java

```diff
@@ -141,6 +144,83 @@ public void setMenuBar(BeanTableFrame<Block> f) {
             ((BlockTableDataModel)m).setDefaultSpeeds(finalF);
         });
         menuBar.add(speedMenu, pos + offset + 1); // put it to the right of the Paths menu
+
+        JMenu valuesMenu = new JMenu(Bundle.getMessage("ValuesMenu"));
+        ButtonGroup valuesButtonGroup = new ButtonGroup();
+        JRadioButtonMenuItem jrbmi = new JRadioButtonMenuItem(Bundle.getMessage("ValuesMenuRestoreAlways"));  // NOI18N
+        jrbmi.addItemListener(new ItemListener() {
+            @Override
+            public void itemStateChanged(ItemEvent e) {
+                setRestoreRule(RestoreRule.RESTOREALWAYS);
+            }
+        });
+        valuesButtonGroup.add(jrbmi);
+        valuesMenu.add(jrbmi);
+        jrbmi.setSelected(_restoreRule == RestoreRule.RESTOREALWAYS);
+
+        jrbmi = new JRadioButtonMenuItem(Bundle.getMessage("ValuesMenuRestoreOccupiedOnly"));  // NOI18N
+        jrbmi.addItemListener(new ItemListener() {
+            @Override
+            public void itemStateChanged(ItemEvent e) {
+                setRestoreRule(RestoreRule.RESTOREOCCUPIEDONLY);
+            }
+        });
+        valuesButtonGroup.add(jrbmi);
+        valuesMenu.add(jrbmi);
+        jrbmi.setSelected(_restoreRule == RestoreRule.RESTOREOCCUPIEDONLY);
+
+        jrbmi = new JRadioButtonMenuItem(Bundle.getMessage("ValuesMenuRestoreOnlyIfAllOccupied"));  // NOI18N
+        jrbmi.addItemListener(new ItemListener() {
+            @Override
+            public void itemStateChanged(ItemEvent e) {
+                setRestoreRule(RestoreRule.RESTOREONLYIFALLOCCUPIED);
+            }
+        });
+        valuesButtonGroup.add(jrbmi);
+        valuesMenu.add(jrbmi);
+        jrbmi.setSelected(_restoreRule == RestoreRule.RESTOREONLYIFALLOCCUPIED);
+
+        menuBar.add(valuesMenu, pos + offset + 2); // put it to the right of the Speed menu
+    
+    }
+
+    /**
+     * Save the restore rule selection. Called by menu item change events.
+     *
+     * @param newRule The RestoreRule enum constant
+     */
+    void setRestoreRule(RestoreRule newRule) {
+        _restoreRule = newRule;
+        InstanceManager.getDefault(jmri.UserPreferencesManager.class).
+                setProperty(getClassName(), "Restore Rule", newRule.toString());  // NOI18N
```

Comment:
I recommend that you use `newRule.name()` instead of `newRule.toString()`. The method `toString()` might get overridden and return something else, like an internationalized name of the rule, while `name()` is final and cannot be overridden. It's guaranteed to always return the name of the enum.

https://docs.oracle.com/javase/8/docs/api/java/lang/Enum.html#name--
https://docs.oracle.com/javase/8/docs/api/java/lang/Enum.html#toString--

Reply:
Thanks, @danielb987 
Fixed.